### PR TITLE
Update mvdr_beamforming FPGA Reference Design to use shared include files

### PR DESCRIPTION
**Pull request by ericxu233**
### Description
Update the FPGA mvdr_beamforming reference design to use code from the shared include directory.

* use shared library resources streaming_qrd.hpp, tuple.hpp, and unrolled_loop.hpp
* updated streaming_qrd.hpp to accept data in a different order, as required by the mvdr design

### Type of change
  
Code cleanup on an existing design.

### How Has This Been Tested?

* The mvdr_beamforming design has been compiled and ran on the FPGA emulator and results are passing.
* The mvdr_beamforming design has been compiled and ran on specific Intel FPGA boards. The executable output of the design shows that the code functionality is passing. 
* QOR (fmax and throughput) of the design was checked on S10 and Agilex boards before and after the change with the large array size, and there were no significant differences

### CODE REVIEW

**Review by mdbtucker — DISMISSED** [review]
Already reviewed in PR to Eric's fork

File: DirectProgramming/DPC++FPGA/include/streaming_qrd.hpp [hunk 1]
```diff
@@ -21,31 +21,37 @@ namespace fpga_linalg {
...
+                            // elements from the pipe with each read
+          typename QOut,    // Q matrix output pipe, send pipe_size
+                            // elements to the pipe with each write
+          typename ROut,    // R matrix output pipe, send pipe_size
+                            // elements to the pipe with each write.
+                            // Only upper-right elements of R are
+                            // sent in row order, starting with row 0.
+          bool k_column_order =
+              true  // Defualt value is true for standard matrix input reads
+                    // (reads the matrix one column at a time). Flase if read
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/include/streaming_qrd.hpp:51]:
```suggestion
                    // (reads the matrix one column at a time). False if read
```

File: DirectProgramming/DPC++FPGA/include/streaming_qrd.hpp [hunk 2]
```diff
@@ -21,31 +21,37 @@ namespace fpga_linalg {
...
+          typename AIn,     // A matrix input pipe, receive pipe_size
+                            // elements from the pipe with each read
+          typename QOut,    // Q matrix output pipe, send pipe_size
+                            // elements to the pipe with each write
+          typename ROut,    // R matrix output pipe, send pipe_size
+                            // elements to the pipe with each write.
+                            // Only upper-right elements of R are
+                            // sent in row order, starting with row 0.
+          bool k_column_order =
+              true  // Defualt value is true for standard matrix input reads
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/include/streaming_qrd.hpp:50]:
```suggestion
              true  // Default value is true for standard matrix input reads
```

File: DirectProgramming/DPC++FPGA/include/streaming_qrd.hpp [hunk 3]
```diff
@@ -210,8 +227,8 @@ struct StreamingQRD {
       // Depending on the context, will contain:
       // -> -s[j]: for all the iterations to compute a_j
       // -> ir: for one iteration per j iterations to compute Q_i
-      [[intel::fpga_memory]]
-      [[intel::private_copies(2)]] // NO-FORMAT: Attribute
+      [[intel::fpga_memory]] [[intel::private_copies(
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/include/streaming_qrd.hpp:230]:
```suggestion
      [[intel::fpga_memory]]      // NO-FORMAT: Attribute
      [[intel::private_copies(2)]] // NO-FORMAT: Attribute
```

File: DirectProgramming/DPC++FPGA/include/streaming_qrd.hpp [hunk 4]
```diff
@@ -210,8 +227,8 @@ struct StreamingQRD {
...
+          2)]]  // NO-FORMAT: Attribute
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/include/streaming_qrd.hpp:231]:
```suggestion

```

**Review by yuguen-intel — CHANGES_REQUESTED** [review]

**Comment by yuguen-intel** [comment]
This change looks very good, I just messaged you @ericxu233 regarding the testing that you did for that change :)
Thanks!

File: DirectProgramming/DPC++FPGA/include/streaming_qrd.hpp [hunk 5]
```diff
@@ -227,8 +227,8 @@ struct StreamingQRD {
       // Depending on the context, will contain:
       // -> -s[j]: for all the iterations to compute a_j
       // -> ir: for one iteration per j iterations to compute Q_i
-      [[intel::fpga_memory]] [[intel::private_copies(
-          2)]]  // NO-FORMAT: Attribute
+      [[intel::fpga_memory]]
```
Comment (yuguen-intel) [inline DirectProgramming/DPC++FPGA/include/streaming_qrd.hpp:230]:
```suggestion
      [[intel::fpga_memory]] // NO-FORMAT: Attribute
```

**Review by yuguen-intel — APPROVED** [review]

**Pull request merged**